File: Online Judges/LeetCode/June LeetCoding Challenge/Week-1/Coin Change 2.py
class Solution:
    def change(self, amount: int, coins: List[int]) -> int:
        dp = [0]*(amount+1)
        dp[0] = 1
        for coin in coins:
            for cur in range(1, amount+1):
                if cur >= coin:
                    dp[cur] += dp[cur-coin]
        return dp[amount]

File: Online Judges/LeetCode/June LeetCoding Challenge/Week-1/Coin Change 2.rs
impl Solution {
    pub fn change(amount: i32, coins: Vec<i32>) -> i32 {
        let amount = amount as usize;

        let mut dp = vec![0; amount +1];
        dp[0] = 1;
        
        for &coin in coins.iter() {
            let coin = coin as usize;    
            for cur in 1..amount+1{
                if(cur >= coin) {
                    dp[cur] += dp[cur - coin];
                }
            }
        }
        dp[amount]
    }
}